Realm/RLMSyncSubscription.mm

```diff
@@ -237,10 +241,45 @@ - (RLMSyncSubscriptionState)state {
 #pragma mark - Batch Update subscriptions
 
 - (void)update:(__attribute__((noescape)) void(^)(void))block {
-    return [self update:block onComplete:nil];
+    [self update:block onComplete:nil];
 }
 
 - (void)update:(__attribute__((noescape)) void(^)(void))block onComplete:(void(^)(NSError *))completionBlock {
+    [self update:block queue:nil onComplete:completionBlock];
+}
+
+// FIXME: Ultimately needs something cancellable from realm-core instead of sdk-level workaround.
+- (void)update:(__attribute__((noescape)) void(^)(void))block
+         queue:(nullable dispatch_queue_t)queue
+       timeout:(NSTimeInterval)timeout
+    onComplete:(void(^)(NSError *))completionBlock {
+    // Create an internal completion block that will only be called once
+    __block BOOL called = false;
```

Comment:
This is a data race. The block can be called on both the main thread and the sync worker thread at the same time if it completes just as it would time out. This also is making it so that the completion block is always captured until the timeout expires.